File: DeductionsBackend/Deductions.Models/Contracts/EmployeeContract.cs
﻿using System.Collections.Generic;

namespace Deductions.Models.Contracts
{
    public class EmployeeContract : BeneficiaryContract
    {
        public EmployeeContract() : base()
        {
            Dependents = new List<DependentContract>();
        }

        public decimal AnnualSalary { get; set; }
        public IEnumerable<DependentContract> Dependents { get; set; }
    }
}

File: deductions/src/app/contracts/employee.contract.ts
import { BeneficiaryContract } from './beneficiary.contract';
import { DependentContract } from './dependent.contract';

export class EmployeeContract extends BeneficiaryContract {
  constructor() {
    super();
    this.annualElectedBenefitsCost = 1000;
  }
  annualSalary = 52000;
  dependents = Array<DependentContract>();
}
